src/main/scala/org/llm4s/LLMConnection.scala

```diff
@@ -0,0 +1,78 @@
+package org.llm4s

```

Comment:
Just realised that this and PricingLoader are copies of each other.
This is closer to the right location ,

I think you can break this into two.

ModelPricingAndContextCache  - reads the file had a 'get method which takes a provider/model pair and returns 'ModelPricingAndContextInstance (sure we can come up with  a btter name) - which then has the the getCost method.

When we create an LLMClient we can look up the cost for the client in the cache if the user has not provided an override.